Source/Diagnostics/BTDiagnostics.cpp

```diff
@@ -481,8 +480,13 @@ BTDiagnostics::DefineCellCenteredMultiFab(int lev)
     ba.coarsen(m_crse_ratio);
     amrex::DistributionMapping dmap = warpx.DistributionMap(lev);
     int ngrow = 1;
+#ifdef WARPX_DIM_RZ
+    int ncomps = WarpX::ncomps * static_cast<int>(m_cellcenter_varnames.size());
+#else
     int ncomps = static_cast<int>(m_cellcenter_varnames.size());
+#endif
     m_cell_centered_data[lev] = std::make_unique<amrex::MultiFab>(ba, dmap, ncomps, ngrow);
+    m_cell_centered_data[lev]->setVal(0.);
```

Comment:
I saw this fix and I think we could try to use here the new function `WarpX::AllocInitMultiFab` implemented in #3484 in order to allocate and initialize to zero the MultiFab `m_cell_centered_data` in one single function call.